lib/Twig/Template.php

```diff
@@ -607,15 +618,23 @@ protected function getAttribute($object, $item, array $arguments = array(), $typ
             $this->env->getExtension('Twig_Extension_Sandbox')->checkMethodAllowed($object, $method);
         }
 
-        // Some objects throw exceptions when they have __call, and the method we try
-        // to call is not supported. If ignoreStrictCheck is true, we should return null.
-        try {
-            $ret = call_user_func_array(array($object, $method), $arguments);
-        } catch (BadMethodCallException $e) {
-            if ($call && ($ignoreStrictCheck || !$this->env->isStrictVariables())) {
-                return;
+        if ($call) {
+            // Some objects throw exceptions when they have __call, and the method we try
+            // to call is not supported. If ignoreStrictCheck is true, we should return null.
+            try {
+                $ret = $object->__call($method, $arguments);
+            } catch (BadMethodCallException $e) {
+                if ($ignoreStrictCheck || !$this->env->isStrictVariables()) {
+                    return;
+                }
+                throw $e;
             }
-            throw $e;
+        } elseif (!$arguments) {
+            $ret = $object->$method();
+        } elseif (PHP_VERSION_ID >= 50600) {
+            $ret = twig_call_method($object, $method, $arguments);
```

Comment:
I don't understand at this point why you're not using unpacking since PHP version, at this point **is** >= 5.6.
However, for 5.[2-5], can't you relay on `twig_call_method`?
Not sure if I'm missing something on this though